src/script/scriptBuilder.zig

```diff
@@ -0,0 +1,429 @@
+        }
+
+        if (dataLen < Opcode.OP_PUSHDATA1.toBytes()) {
+            return 1 + dataLen;
+        } else if (dataLen <= 0xff) {
+            return 2 + dataLen;
+        } else if (dataLen <= 0xffff) {
+            return 3 + dataLen;
+        }
+
+        return 5 + dataLen;
+    }
+
+    /// Private Function to addData without checking MAX_SCRIPT_SIZE
+    /// Only to be used for testing soundess of the ScriptBuilder struct
+    /// Cannot be called from other files
+    fn addDataUnchecked(self: *ScriptBuilder, data: []const u8) !*ScriptBuilder {
+        if (data.len == 0 or data.len == 1 and data[0] == 0) {
+            _ = try self.addOpcode(Opcode.OP_0);
+        } else if (data.len == 1 and data[0] <= 16) {
+            const op = Opcode.OP_1.toBytes() - 1 + data[0];
+            try self.script.append(op);
+        } else if (data.len == 1 and data[0] == 0x81) {
+            _ = try self.addOpcode(Opcode.OP_1NEGATE);
+        } else if (data.len < Opcode.OP_PUSHDATA1.toBytes()) {
+            try self.script.append(@intCast(data.len));
+            try self.script.appendSlice(data);
+        } else if (data.len <= 0xff) {
+            try self.script.append(Opcode.OP_PUSHDATA1.toBytes());
+            try self.script.append(@intCast(data.len));
+
+            try self.script.appendSlice(data);
+        } else if (data.len <= 0xffff) {
+            try self.script.append(Opcode.OP_PUSHDATA2.toBytes());
+            const length: u16 = @intCast(data.len);
+            var n: [2]u8 = undefined;
+            std.mem.writeInt(u16, &n, length, .little);
+            try self.script.appendSlice(&n);
+
+            try self.script.appendSlice(data);
+        } else {
+            try self.script.append(Opcode.OP_PUSHDATA4.toBytes());
+            const length: u32 = @intCast(data.len);
+            var n: [4]u8 = undefined;
+            std.mem.writeInt(u32, &n, length, .little);
+
+            try self.script.appendSlice(&n);
+            try self.script.appendSlice(data);
+        }
+        return self;
+    }
+};
+
+test "ScriptBuilder Smoke test" {
+    var sb = try ScriptBuilder.new(std.testing.allocator, 3);
+    defer sb.deinit();
+
+    var e = try (try (try (try sb.addInt(1)).addInt(2)).addOpcode(Opcode.OP_ADD)).build();
+    const expected = [_]u8{ 0x51, 0x52, 0x93 };
+    defer e.deinit();
+
+    try std.testing.expectEqualSlices(u8, &expected, e.script.data);
+}
+
+//METHOD 1
+test "ScriptBuilder OP_SWAP METHOD 1" {
+    var sb = try ScriptBuilder.new(std.testing.allocator, 4);
+
+    var e = try (try (try (try (try sb.addInt(1)).addInt(2)).addInt(3)).addOpcode(Opcode.OP_SWAP)).build();
+
+    const expected = [_]u8{ 0x51, 0x52, 0x53, 0x7c };
+
+    defer sb.deinit();
+    defer e.deinit();
+
+    try std.testing.expectEqualSlices(u8, &expected, e.script.data);
+}
+//METHOD 2
+test "ScriptBuilder OP_SWAP METHOD 2" {
+    var sb = try ScriptBuilder.new(std.testing.allocator, 4);
+
+    defer sb.deinit();
```

Comment:
always put the defer on the line just below the declaration. No empty lane between those

Reply:
It make it much more readable